wrappers/fiber/fiber.go

```diff
@@ -0,0 +1,190 @@
+package epsagonfiber
+
+import (
+	"encoding/json"
+	"fmt"
+	"runtime/debug"
+	"strconv"
+
+	"github.com/epsagon/epsagon-go/epsagon"
+	"github.com/epsagon/epsagon-go/protocol"
+	"github.com/epsagon/epsagon-go/tracer"
+	"github.com/gofiber/fiber/v2"
+	"github.com/valyala/fasthttp"
+)
+
+// GinRouterWrapper is an epsagon instumentation wrapper for gin.RouterGroup
+type FiberEpsagonMiddleware struct {
+	Config *epsagon.Config
+}
+
+func parseQueryArgs(args *fasthttp.Args) map[string]string {
+	result := make(map[string]string)
+	args.VisitAll(func(key, val []byte) {
+		result[string(key)] = string(val)
+	})
+	return result
+}
+
+// convert map values to string. On error, add exception to tracer with the
+// given error message
+func convertMapValuesToString(
+	values map[string]string,
+	wrapperTracer tracer.Tracer,
+	errorMessage string) string {
+	processed, err := json.Marshal(values)
+	if err != nil {
+		wrapperTracer.AddException(&protocol.Exception{
+			Type:      "trigger-creation",
+			Message:   errorMessage,
+			Traceback: string(debug.Stack()),
+			Time:      tracer.GetTimestamp(),
+		})
+		return ""
+	}
+	return string(processed)
+}
+
+func processQueryFromURI(uriObj *fasthttp.URI, wrapperTracer tracer.Tracer) string {
+	if uriObj == nil {
+		return ""
+	}
+	args := parseQueryArgs(uriObj.QueryArgs())
+	return convertMapValuesToString(
+		args,
+		wrapperTracer,
+		fmt.Sprintf("Failed to serialize query params %s", uriObj.QueryString()))
+
+}
+
+func processRequestHeaders(requestHeaders *fasthttp.RequestHeader, wrapperTracer tracer.Tracer) string {
+	if requestHeaders == nil {
+		return ""
+	}
+	headers := make(map[string]string)
+	requestHeaders.VisitAll(func(key, val []byte) {
+		headers[string(key)] = string(val)
+	})
+	return convertMapValuesToString(
+		headers,
+		wrapperTracer,
+		fmt.Sprintf("Failed to serialize request headers"))
+}
+
+func processResponseHeaders(responseHeaders *fasthttp.ResponseHeader, wrapperTracer tracer.Tracer) string {
+	if responseHeaders == nil {
+		return ""
+	}
+	headers := make(map[string]string)
+	responseHeaders.VisitAll(func(key, val []byte) {
+		headers[string(key)] = string(val)
+	})
+	return convertMapValuesToString(
+		headers,
+		wrapperTracer,
+		fmt.Sprintf("Failed to serialize response headers"))
+
+}
+
+// Gets the request content type header, empty string if not found
+func getRequestContentType(fiberCtx *fiber.Ctx) string {
+	return string(fiberCtx.Request().Header.ContentType())
+}
+
+// CreateHTTPTriggerEvent creates an HTTP trigger event
+func CreateHTTPTriggerEvent(wrapperTracer tracer.Tracer, fiberCtx *fiber.Ctx, resourceName string) *protocol.Event {
+	request := fiberCtx.Request()
+
+	name := resourceName
+	if len(name) == 0 {
+		name = fiberCtx.Hostname()
+	}
+
+	event := &protocol.Event{
+		Id:        "",
```

Comment:
Shouldn't we generate an id, or is that done somewhere else?

Reply:
it will be done automatically if not given